example/App.tsx

```diff
@@ -1,54 +1,73 @@
-import * as React from 'react';
-import { StyleSheet, View, Text, Image, Animated, Button } from 'react-native';
+import React, { useRef, useState } from 'react';
+import { View, Text, Image, Animated, Button } from 'react-native';
 // @ts-ignore
 import { ReactNativeZoomableView } from '@openspacelabs/react-native-zoomable-view';
+import { styles } from './style';
 
 export default function App() {
-  const zoomAnimatedValue = React.useRef(new Animated.Value(1)).current;
+  const zoomAnimatedValue = useRef(new Animated.Value(1)).current;
   const scale = Animated.divide(1, zoomAnimatedValue);
-  const [showMarkers, setShowMarkers] = React.useState(true);
+  const [showMarkers, setShowMarkers] = useState(true);
+  const [size, setSize] = useState<{ width: number; height: number }>();
+
+  // Use layout event to get centre point, to set the pin
+  const [pin, setPin] = useState({ x: 0, y: 0 });
+  // Debounce the change event to avoid layout event firing too often while dragging
+  const debouncedUpdatePin = debounce(setPin, 10);
```

Comment:
Creating a new debounce function on every render will cause the internal debounce context to reset every time.

I think we can import `useDebounce` into the example package.json and do
```ts
const debouncedPin = useDebounce(pin, 10)
```

Reply:
Trickier than that - setting the pin is what we want to debounce. I found an awkward way with 

```tsx
const debouncedUpdatePin = useCallback(() => debounce(setPin, 10), [])();
```